.github/workflows/terraform-observe_scheduler.yaml

```diff
@@ -27,33 +32,55 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v4
-      - name: Build matrix
-        id: matrix
-        shell: bash
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+
+      - name: download python script
+        id: matrix_v2
         run: |
-          DIRS="$(find . -path "*tftests/*" ! -path "*.terraform*" -type d )"
-          [[ ${#DIRS} > 0 ]] && echo "directories=[\"${DIRS//$'\n'/\",\"}\"]" >> "$GITHUB_OUTPUT" || echo "directories=[]" >> "$GITHUB_OUTPUT"
+          curl -O https://raw.githubusercontent.com/observeinc/.github/${{ inputs.script_branch }}/.github/scripts/main.py
+          # script creates output directories_with_customerID
+          python3 main.py
+        env:
+          TERRAFORM_MODULES_TEST_OBSERVE_CUSTOMER_LIST: ${{ secrets.TERRAFORM_MODULES_TEST_OBSERVE_CUSTOMER_LIST }}
+
     outputs:
-      directories: ${{ steps.matrix.outputs.directories }}
+      directories_with_customerID: ${{ steps.matrix_v2.outputs.directories_with_customerID }}
 
   terraform-test-apply:
     needs: get-test-directories
-    if: ${{ needs.get-test-directories.outputs.directories != '[]' && needs.get-test-directories.outputs.directories != '' }}
+    # if: ${{ needs.get-test-directories.outputs.directories_with_customerID != '' }}
     name: Test Terraform Module
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
+      max-parallel: 2
       matrix:
-        directory: ${{ fromJson(needs.get-test-directories.outputs.directories) }}
+        directories_with_customerID: ${{ fromJson(needs.get-test-directories.outputs.directories_with_customerID) }}
     steps:
       - uses: hashicorp/setup-terraform@v3
         with:
           terraform_version: ${{ inputs.terraform-version }}
       - uses: actions/checkout@v4
+
+      - name: split string
+        run: |
+          # Define the string
+          string="${{ matrix.directories_with_customerID }}"
+          # Split the string using awk
+          directory=$(awk -F '_XXXXXX_' '{print $1}' <<< "$string")
+          customerID=$(awk -F '_XXXXXX_' '{print $2}' <<< "$string")
+          echo "TEST_DIR=$directory" >> $GITHUB_ENV
+          echo "CUST_NUMBER=$customerID" >> $GITHUB_ENV
```

Comment:
Is it possible to move this step before the terraform step ? 
Also can this be put in the Python script too, as from what I understand, you're creating more env vars?

Reply:
Unless this is for visibility to keep it in the workflow file? Then it's probably fine